db/migrate/20190108031411_add_image_name_to_operating_systems.rb

```diff
@@ -0,0 +1,157 @@
+class AddImageNameToOperatingSystems < ActiveRecord::Migration[5.0]
+  include MigrationHelper
+
+  StubOperatingSystemObj = Struct.new(:operating_system) do
+    attr_reader :hardware # always nil
+    def name
+      ""
+    end
+  end
+
+  class OperatingSystem < ActiveRecord::Base
+    belongs_to :host
+    belongs_to :vm_or_template
+    belongs_to :computer_system
+
+    # Using a `before_save` here since this is the mechanism that will be used
+    # in the app.  Causes a bit of issues in the specs, but proves that this
+    # would work moving forward.
+    before_save :update_platform_and_image_name
+
+    def update_platform_and_image_name
+      obj = case
```

Comment:
```suggestion
      obj = host || vm_or_template || computer_system || StubOperatingSystemObj.new(self)
```

Reply:
But is that under 80 characters?

Reply:
close? 84 (not counting the indentation)